test/service_manager_test.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:async';
+import 'package:devtools/eval_on_dart_library.dart';
+import 'package:devtools/globals.dart';
+import 'package:devtools/service_manager.dart';
+import 'package:devtools/vm_service_wrapper.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+import 'package:vm_service_lib/vm_service_lib.dart';
+
+import 'test_driver.dart';
+
+void main() {
+  group('serviceManagerTests', () {
+    FlutterRunTestDriver _flutter;
+    VmServiceWrapper service;
+
+    setUp(() async {
+      _flutter =
+          FlutterRunTestDriver(fs.directory('test/fixtures/flutter_app'));
+
+      await _flutter.run(withDebugger: true);
+      service = _flutter.vmService;
+
+      setGlobal(ServiceConnectionManager, new ServiceConnectionManager());
+
+      final completer = new Completer<Null>();
+      await serviceManager.vmServiceOpened(service, completer.future);
+    });
+
+    tearDown(() async {
+      await service.allFuturesCompleted.future;
+      _flutter.stop();
+    });
+
+    test('vmServiceOpened', () async {
+      expect(serviceManager.service, equals(service));
+      expect(serviceManager.isolateManager, isNotNull);
+      expect(serviceManager.serviceExtensionManager, isNotNull);
+      expect(serviceManager.isolateManager.isolates, isNotEmpty);
+
+      // Test should time out and fail if the following await statements do not
```

Comment:
This comment can be deleted as it doesn't provide any clear value.

Reply:
Done.